x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/aggs/frequentitemsets/ItemSetTraverser.java

```diff
@@ -132,11 +156,25 @@ public void prune() {
             return;
         }
         itemIdStack.length--;
+        itemPositionsStack.length--;
+        itemPositionsVectorParent.clear(itemPositionsStack.ints[itemPositionsStack.length]);
+        itemPositionsVector.clear(itemPositionsStack.ints[itemPositionsStack.length]);
     }
 
-    @Override
-    public void close() {
-        Releasables.close(topItemIds);
+    private void growStacksIfNecessary() {
+        if (itemIdStack.longs.length == itemIdStack.length) {
+            LongsRef resizedItemIdStack = new LongsRef(itemIdStack.length + SIZE_INCREMENT);
+            System.arraycopy(itemIdStack.longs, 0, resizedItemIdStack.longs, 0, itemIdStack.length);
```

Comment:
I know this is simply a code move, but can we use `ArrayUtil.grow` on the underlying arrays and create a new ref?

Reply:
Goes for both refs.